controllers/auth.js

```diff
@@ -122,9 +127,32 @@ const updateAuthStatus = async (req, res) => {
   }
 };
 
+const fetchUserDeviceInfo = async (req, res) => {
+  try {
+    const queryParamArray = Object.keys(req.query);
+    if (queryParamArray.length === 1 && queryParamArray[0] === "device_id") {
+      const deviceId = req.query.device_id;
+      const userDeviceInfoData = await QrCodeAuthModel.retrieveUserDeviceInfo(deviceId);
+      if (!userDeviceInfoData.userExists) {
+        return res.boom.notFound("No Authentication found!");
+      }
```

Comment:
error message is very generic

Reply:
Replace the message with "User with id <:id> does not exist."